### DISCUSSION TITLE
Dynamic Attribute Names

### DISCUSSION THREAD

**Discussion by dartanian300** [lit/lit#2395] [Q&A]
Hello all - fairly new to Lit and trying to slot it into an existing legacy application. Is it possible to dynamically set the name of an attribute? I have an object of attributes similar to this:

`{
  weight: 900,
  size: 14,
}`

These property names are arbitrary and can be anything. I need to use them as attribute names. The output would be like:

`<div data-weight="900" data-size="14">...</div>`

From what I can tell this isn't possible in lit. Am I correct?

**Comment by justinfagnani** [comment]
You could do this with a custom directive. Something like:
```ts
class SetAttr extends Directive {
  render(name: string, value: string) {
    // there's no Lit value we can return to set an arbitrary attribute
  }
  update(part: ElementPart, [name, value]: DirectiveParameters<this>) {
    part.element.setAttribute(name, value);
  }
}
export const setAttr = directive(SetAttr);
```
